CMakeLists.txt

```diff
@@ -24,6 +30,25 @@ option(BUILD_DOXYGEN "Enable/Disable the generation of the Doxygen." OFF)
 option(BUILD_EXAMPLES "Enable/Disable the compilation of the examples." OFF)
 option(BUILD_TESTS "Enable/Disable the compilation of unit tests." ON)
 option(BUILD_TOOLS "Enable/Disable the compilation of tools." ON)
+option(BUILD_SHARED_LIBS "Enable/disable build of shared libraries." ON)
+option(BUILD_SHARED_EXTENSIONS "Enable/disable build of shared extensions." ON)
+option(BUILD_STATIC_LIBS "Enable/disable build of static libraries." ON)
+option(BUILD_STATIC_EXTENSIONS "Enable/disable build of static extensions." ON)
+option(MSVC_STATIC_RUNTIME_LIBRARY "Enable/disable static runtime on Windows." OFF)
+option(BOOST_STATIC_LIBS "Use static version of boost libraries." OFF)
+
+if (BOOST_STATIC_LIBS AND BUILD_SHARED_LIBS)
+    set(BOOST_STATIC_LIBS OFF)
+    message(WARNING "BOOST_STATIC_LIBS disabled if you want to build shared libraries. Use this option with BUILD_SHARED_LIBS=OFF")
+endif ()
+
+if (BUILD_TESTS AND NOT BUILD_SHARED_LIBS)
+    message(FATAL_ERROR "BUILD_TESTS is ON but BUILD_SHARED_LIBS is OFF, you should turn ON BUILD_SHARED_LIBS or turn off BUILD_TESTS.")
+endif ()
```

Comment:
Maybe we have to do something similar for the examples. Maybe we can find a way to link the examples and the tests to the IIDM library, using a postfix for example (empty or -static).